src/software/util/sml_fsm/sml_fsm.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <functional>
+#include <include/boost/sml.hpp>
+#include <queue>
+
+/**
+ * The Tactic FSM framework uses the [SML library](https://github.com/boost-ext/sml), and
+ * aims to create a readable style of FSM to implement tactic gameplay. See the MoveTactic
+ * for an example of how to implement a tactic using this framework
+ */
+
+// An alias for an FSM
+template <class T>
+using FSM = boost::sml::sm<T, boost::sml::process_queue<std::queue>>;
+
+/**
+ * Defines an sml state wrapper around a class/struct
+ *
+ * @param STATE The state class/struct
+ */
+#define DEFINE_SML_STATE(STATE) const auto STATE##_S = boost::sml::state<STATE>;
+
+/**
+ * Defines an sml event wrapper around a class/struct
```

Comment:
nit: "sml" and "SML" are being used interchangeably in this file's comments, probably just use one or the other

Reply:
fixed